WordPress/Classes/ViewRelated/Post/PostListViewController.swift

```diff
@@ -0,0 +1,685 @@
+import Foundation
+import WordPressComAnalytics
+import WordPressComStatsiOS
+import WordPressShared
+
+@objc class PostListViewController : AbstractPostListViewController, UIViewControllerRestoration, PostCardTableViewCellDelegate {
+    
+    static private let postCardTextCellIdentifier = "PostCardTextCellIdentifier"
+    static private let postCardImageCellIdentifier = "PostCardImageCellIdentifier"
+    static private let postCardRestoreCellIdentifier = "PostCardRestoreCellIdentifier"
+    static private let postCardTextCellNibName = "PostCardTextCell"
+    static private let postCardImageCellNibName = "PostCardImageCell"
+    static private let postCardRestoreCellNibName = "RestorePostTableViewCell"
+    static private let postsViewControllerRestorationKey = "PostsViewControllerRestorationKey"
+    static private let statsStoryboardName = "SiteStats"
+    static private let currentPostListStatusFilterKey = "CurrentPostListStatusFilterKey"
+    static private let currentPostAuthorFilterKey = "CurrentPostAuthorFilterKey"
+    
+    // TODO: low cap on first char!
+    
+    static private let statsCacheInterval = NSTimeInterval(300) // 5 minutes
+    static private let postCardEstimatedRowHeight = Float(100.0)
+    static private let postCardRestoreCellRowHeight = Float(54.0)
+    static private let postListHeightForFooterView = CGFloat(34.0)
+    
+    @IBOutlet private var textCellForLayout : PostCardTableViewCell!
+    @IBOutlet private var imageCellForLayout : PostCardTableViewCell!
+    @IBOutlet private weak var authorFilterSegmentedControl : UISegmentedControl!
+    
+    // MARK: Initializers & deinitializers
+    
+    deinit {
+        PrivateSiteURLProtocol.unregisterPrivateSiteURLProtocol()
+    }
+    
+    required init?(coder aDecoder: NSCoder) {
+        super.init(coder: aDecoder)
+        
+        PrivateSiteURLProtocol.registerPrivateSiteURLProtocol()
+    }
+    
+    // MARK: - Convenience constructors
+    
+    class func controllerWithBlog(blog: Blog) -> PostListViewController {
+        
+        let storyBoard = UIStoryboard(name: "Posts", bundle: NSBundle.mainBundle())
+        let controller = storyBoard.instantiateViewControllerWithIdentifier("PostListViewController") as! PostListViewController
```

Comment:
I've got second thoughts on whether we wanna stick to a Storyboard, or not. There are just two viewControllers there, and (if we) switched to Nibs, this method could just turn into an initializer.

But that's definitely another issue! =D